### PR TITLE
create a navigation menu component for the landing page

### PR DESCRIPTION
**Pull request by vithushan19**
This PR adds a navigation component from the https://ui.shadcn.com/docs library.

### CODE REVIEW

File: components/NavigationMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import * as React from "react";
+import * as NavigationMenuPrimitive from "@radix-ui/react-navigation-menu";
+import { cva } from "class-variance-authority";
+import { ChevronDown } from "lucide-react";
+
+import { cn } from "../lib/utils";
+
+const NavigationMenu = React.forwardRef<
```
Comment (vithushan19) [inline components/NavigationMenu.tsx:8]: copied this from the shadcdn components

File: components/coding/landing/Hero.tsx [hunk 2]
```diff
@@ -1,7 +1,5 @@
-import Link from "next/link";
```
Comment (vithushan19) [inline components/coding/landing/Hero.tsx:1]: this file is just fixing lint warnings

File: components/coding/landing/SubscribeBanner.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from "react";
```
Comment (vithushan19) [inline components/coding/landing/SubscribeBanner.tsx:1]: lint warning

File: lib/utils.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import clsx, { ClassValue } from "clsx";
+import { twMerge } from "tailwind-merge";
+
+export function cn(...inputs: ClassValue[]) {
```
Comment (vithushan19) [inline lib/utils.ts:4]: copied from shadcdn components

File: components/NavigationMenu.tsx (hunk 1 above)
Comment (kwiedenhaupt) [reply components/NavigationMenu.tsx:8]: I would love to hear you explain the forwardRef more tomorrow hopefully.  I am coming across this in the radix-ui library and want to make sure I am understanding it properly.

File: components/NavigationMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    )}
+    {...props}
+  />
+));
+NavigationMenuList.displayName = NavigationMenuPrimitive.List.displayName;
+
+const NavigationMenuItem = NavigationMenuPrimitive.Item;
+
+const navigationMenuTriggerStyle = cva(
+  "inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus:outline-none focus:bg-slate-100 disabled:opacity-50 dark:focus:bg-slate-800 disabled:pointer-events-none bg-transparent hover:bg-slate-100 dark:hover:bg-slate-800 dark:text-slate-100 dark:hover:text-slate-100 data-[state=open]:bg-slate-50 dark:data-[state=open]:bg-slate-800 h-10 py-2 px-4 group w-max"
```
Comment (kwiedenhaupt) [inline components/NavigationMenu.tsx:44]: I also see that you use data-[state=open].  I did read that the newest version of tailwind supports this, but not some of the earlier versions.  Did you have to update?  In working on my sandbox for radix-ui, I did update my version of tailwind to 3.2.7 from 3.1.8 and didn't have any issues.  Just curious if you updated as well.  I did a default open with a tooltip and a modal on test and it seemed to work.

**Comment by vithushan19** [comment]
we're not using shadcn components anymore.

**Pull request closed without merging**